src/Features/CloudShadows.cpp

```diff
@@ -0,0 +1,283 @@
+			if (rtvs[0] == reflections.cubeSideRTV[i]) {
+				side = i;
+				break;
+			}
+		if (side == -1)
+			return;
+
+		CheckResourcesSide(side);
+
+		rtvs[3] = cubemapCloudOccRTVs[side];
+		context->OMSetRenderTargets(4, rtvs, depthStencil);
+
+		// blend states
+		ID3D11BlendState* blendState;
+		FLOAT blendFactor[4];
+		UINT sampleMask;
+
+		context->OMGetBlendState(&blendState, blendFactor, &sampleMask);
+
+		if (!mappedBlendStates.contains(blendState)) {
+			if (!modifiedBlendStates.contains(blendState)) {
+				D3D11_BLEND_DESC blendDesc;
+				blendState->GetDesc(&blendDesc);
+
+				blendDesc.RenderTarget[3] = blendDesc.RenderTarget[0];
+
+				ID3D11BlendState* modifiedBlendState;
+				DX::ThrowIfFailed(device->CreateBlendState(&blendDesc, &modifiedBlendState));
+
+				mappedBlendStates.insert(modifiedBlendState);
+				modifiedBlendStates.insert({ blendState, modifiedBlendState });
+			}
+			blendState = modifiedBlendStates[blendState];
+			context->OMSetBlendState(blendState, blendFactor, sampleMask);
+		}
+	}
+}
+
+void CloudShadows::ModifyLighting()
+{
+	auto context = RE::BSGraphics::Renderer::GetSingleton()->GetRuntimeData().context;
+
+	auto shadowState = RE::BSGraphics::RendererShadowState::GetSingleton();
+	auto cubeMapRenderTarget = !REL::Module::IsVR() ? shadowState->GetRuntimeData().cubeMapRenderTarget : shadowState->GetVRRuntimeData().cubeMapRenderTarget;
+	if (cubeMapRenderTarget != RE::RENDER_TARGETS_CUBEMAP::kREFLECTIONS) {
+		static FrameChecker frame_checker;
+		if (frame_checker.isNewFrame())
+			context->GenerateMips(texCubemapCloudOcc->srv.get());
+
+		auto srv = texCubemapCloudOcc->srv.get();
+		context->PSSetShaderResources(40, 1, &srv);
+	} else {
+		ID3D11ShaderResourceView* srv = nullptr;
+		context->PSSetShaderResources(40, 1, &srv);
+	}
+
+	ID3D11ShaderResourceView* views[1]{};
+	views[0] = perPass->srv.get();
+	context->PSSetShaderResources(23, ARRAYSIZE(views), views);
+}
+
+void CloudShadows::Draw(const RE::BSShader* shader, const uint32_t descriptor)
+{
+	static FrameChecker frame_checker;
+	if (frame_checker.isNewFrame()) {
+		// update settings buffer
+		auto context = RE::BSGraphics::Renderer::GetSingleton()->GetRuntimeData().context;
+
+		PerPass perPassData{};
+
+		perPassData.Settings = settings;
+		perPassData.Settings.TransparencyPower = exp2(perPassData.Settings.TransparencyPower);
+		perPassData.RcpHPlusR = 1.f / (settings.CloudHeight + settings.PlanetRadius);
+
+		D3D11_MAPPED_SUBRESOURCE mapped;
+		DX::ThrowIfFailed(context->Map(perPass->resource.get(), 0, D3D11_MAP_WRITE_DISCARD, 0, &mapped));
+		size_t bytes = sizeof(PerPass);
+		memcpy_s(mapped.pData, bytes, &perPassData, bytes);
+		context->Unmap(perPass->resource.get(), 0);
+	}
+
+	switch (shader->shaderType.get()) {
+	case RE::BSShader::Type::Sky:
+		ModifySky(shader, descriptor);
+		break;
+	case RE::BSShader::Type::Lighting:
+	case RE::BSShader::Type::DistantTree:
+	case RE::BSShader::Type::Grass:
+		// case RE::BSShader::Type::Water:
```

Comment:
remove dead code?

Reply:
I don't think it is dead, just quiescence. Also refer to https://github.com/doodlum/skyrim-community-shaders/pull/172#discussion_r1476479931